src/DataStructures/Tensor/EagerMath/DeterminantAndInverse.hpp

```diff
@@ -368,3 +369,49 @@ auto determinant_and_inverse(
   return determinant_and_inverse_detail::DetAndInverseImpl<
       Symm, Index0, Index1>::apply(tensor);
 }
+
+/*!
+ * \ingroup TensorGroup
+ * \brief Computes the determinant and inverse of a rank-2 Tensor.
+ *
+ * Computes the determinant and inverse together, because this leads to fewer
+ * operations compared to computing the determinant independently.
+ *
+ * \param tensor the input rank-2 Tensor.
+ * \tparam DetTag the Tag for the determinant of input Tensor.
+ * \tparam InvTag the Tag for the inverse of input Tensor.
+ * \return a `Variables` that holds the determinant and inverse of the input
+ * tensor, tagged by `DetTag` and `InvTag` respectively.
+ *
+ * \details
+ * See determinant_and_inverse().
+ */
+template <typename DetTag, typename InvTag, typename T, typename Symm,
+          typename Index0, typename Index1>
+auto determinant_and_inverse(
+    const Tensor<T, Symm, tmpl::list<Index0, Index1>>& tensor) noexcept
+    -> Variables<tmpl::list<DetTag, InvTag>> {
+  static_assert(cpp17::is_same_v<typename DetTag::type, Scalar<T>>,
```

Comment:
In general `typename DetTag::type` -> `db::item_type<DetTag>`. Getting the type from a tag using `::type` is not a supported interface for DataBox tags. In this case it's "okay" since the tag is used in a `Variables` which requires this be a valid way of retrieving the type.

I think once all the GH PRs currently submitted are in it would be good to do a GH and GR code cleanup making sure all the dox renders correctly, things like `::type` aren't used, etc.